maro/simulator/scenarios/data_center/business_engine.py

```diff
@@ -0,0 +1,204 @@
+from copy import deepcopy
+from typing import List
+
+from yaml import safe_load
+
+from maro.event_buffer import AtomEvent, CascadeEvent, EventBuffer, MaroEvents
+from maro.simulator.scenarios.abs_business_engine import AbsBusinessEngine
+from maro.simulator.scenarios.helpers import DocableDict
+
+from .common import Action, Latency, Payload
+from .events import Events
+from .physical_machine import PhysicalMachine
+from .virtual_machine import VirtualMachine
+
+metrics_desc = """
+
+energy_consumption (int): Current total energy consumption.
+success_requirements (int): Accumulative successful VM requirements until now.
+total_latency (int): Accumulative buffer time until now.
+"""
+
+
+class DataCenterBusinessEngine(AbsBusinessEngine):
+    def __init__(
+        self, event_buffer: EventBuffer, topology: str, start_tick: int,
+        max_tick: int, snapshot_resolution: int, max_snapshots: int, additional_options: dict = {}
+    ):
+        super().__init__(
+            "data_center", event_buffer, topology, start_tick, max_tick,
+            snapshot_resolution, max_snapshots, additional_options
+        )
+
+        self._energy_consumption: int = 0
+        self._success_requirements: int = 0
+        self._total_latency: Latency = Latency()
+
+        self._init()
+
+    def _load_configs(self):
+        """Load configurations"""
+        with open(os.path.join(self._config_path, "config.yml")) as fp:
+            self._conf = safe_load(fp)
+
+    def _init(self):
+        # load config
+        self._load_configs()
+        self._register_events()
+        # PM init
+        self.machines: List[PhysicalMachine] = []
+        pm_amount = self._conf["pm_amount"]
+        cap_cpu = self._conf["pm_cap_cpu"]
+        cap_mem = self._conf["pm_cap_mem"]
+        self.machines = [PhysicalMachine(id=i, cap_cpu=cap_cpu, cap_mem=cap_mem) for i in range(pm_amount)]
+        self.vm: dict = {}
+
+    def step(self, tick: int):
+        """
+        Push business to next step.
+
+        Args:
+            tick (int): Current tick
+        """
+        # Load VM info into payload
+
+        # Update all live VMs memory utilization
+        self._update_vm_util(tick)
+        # Update all PM memory utilization
+        self._update_pm_util(tick)
+        # VM requirement event
+        vm_required_evt = self._event_buffer.gen_cascade_event(tick, Events.REQUIREMENTS, payload=None)
+        self._event_buffer.insert_event(vm_required_evt)
+
+    def get_metrics(self) -> dict:
+        """Get current enviornment metrics information.
+
+        Args:
+            dict: A dict that contains the environment metrics.
+
+        Returns:
+            dict: Metrics information.
+        """
+
+        energy_consumption = self._energy_consumption
+        success_requirements = self._success_requirements
+        total_latency = self._total_latency
+        return DocableDict(
+            metrics_desc,
+            energy_consumption=energy_consumption,
```

Comment:
I'm not sure which one is better, directly using ``energy_consumption=self._energy_consumption`` or the current one.

Do you have any reference which one is better? We can decide and follow the same pattern in the implementation of this scenario